### ISSUE TITLE
Chrome 63 is ignoring autocomplete="off"

### ISSUE THREAD

**Issue by ibarral** [jackocnr/intl-tel-input#668]
Chrome 63 is ignoring autocomplete="off", and tries to suggest previous user entries.

The only solution is to assign a random string to the attribute, for example:

`autocomplete="nope"`

Source: https://developer.mozilla.org/en-US/docs/Web/Security/Securing_your_site/Turning_off_form_autocompletion

I think that we'll need a fix to that :(

**Comment by jackocnr** [comment]
Thanks for raising this. As I mentioned, I would be open to a PR for this tweak! Thanks.

**Comment by woeterman94** [comment]
Any update on this?
Can we still use autocomplete=off on input fields?

**Comment by woeterman94** [comment]
Solution: use autocomplete="nope"

**Comment by jackocnr** [comment]
@woeterman94 yup, that was in the original post haha. I'm open to a pull request :)

**Comment by woeterman94** [comment]
Created one :)

**Comment by jackocnr** [comment]
Strangely, on Chrome v67 (67.0.3396.79) on MacOS, having the autocomplete attribute set to "off" is working for me now (disabling autocompletion), and setting it to "nope" is not working (autocomplete is working again). Are you guys seeing the same thing?

**Comment by woeterman94** [comment]
Works for me with the latest chrome build on windows.

**Comment by jackocnr** [comment]
@ibarral what's your experience? (and are you Windows/Mac etc?)

**Comment by tinisasa** [comment]
I'm still having this issue on inputs, forms, using autocomplete"off/nope/whatever" and/or autofill"off/nope/whatever" 
Chrome Version 66.0.3359.181 (Official Build) (64-bit)
MacOD 10.13.3

**Comment by Shadosky** [comment]
It look like you can use `autocomplete="new-password"` now. 

cf : https://bugs.chromium.org/p/chromium/issues/detail?id=370363#c7

Tested on Version 67.0.3396.99 (Build officiel) (64 bits), & working

**Comment by jackocnr** [comment]
@Shadosky unfortunately that suggestion was from 4 years ago! I'm closing this for now, as it seems it is impossible to fix!

**Comment by juan0087** [comment]
I'm on Chrome Version 68.0.3440.106 (Official Build) (64-bit) and I was able to hide it using       autoComplete="nope" but only on mac osx. On the same browser in a Windows 10 machine it still shows the autocomplete.

**Comment by justjuank** [comment]
I was previously using autocomplete="none" and it was working fine before two weeks ago. Seems like with Chrome Version 68.0.3440.106 they fixed it, I just tried autocomplete="off" and it disables the auto suggestion correctly.

**Comment by nasir-hashe** [comment]
Nope!  its not working in **Version 72.0.3626.109 (Official Build) (64-bit)**. To verify you the issue please see video. https://www.screencast.com/t/2ELzcetLI

In this video I have already set autocomplete="off" on form tag and input box.

**Comment by adrianbj** [comment]
@jackocnr - this is a real problem still. What is currently working for me is hacking the core to use a random string, eg:

```
this.a.setAttribute("autocomplete",Math.random().toString(36).replace(/[^a-z]+/g, '').substr(0, 5));
```

Lots of other ways to do this, but this seems to work ok for now.

**Comment by adrianbj** [comment]
Or maybe just let us handle it ourselves - adding it via this library makes it override our own attempts to manually set it.

**Comment by justjuank** [comment]
Yes, the autocomplete attribute is still a problem. 
I'm not using this library but my current workaround is to set a random name on my input with autocomplete="off" and before submitting the form I set the name attribute back to what it should be.
Yes, an ugly hack but it works so far.

**Comment by jackocnr** [comment]
@adrianbj 

> adding it via this library makes it override our own attempts to manually set it

Can't you just apply your own technique after initialising the plugin?

**Comment by adrianbj** [comment]
@jackocnr - yeah, we probably could do that, although it I was initially confused why I couldn't set autocomplete via `autocomplete="whateveriwant"` in the input tag because the JS in the library overrides that. We'd have to override it via our own JS. I guess I also don't see the point to adding the `autocomplete="off"` at all - it is completely useless these days.

Unfortunately, even my random string approach is not perfect because if the user enters something, then deletes it, autocomplete will kick in again and try to complete what they previously just had entered. I feel like there needs to be a `keyup` or similar eventlistener that dynamically changes the autocomplete value.

What a waste of time this whole thing is :)

**Comment by jackocnr** [comment]
> I guess I also don't see the point to adding the `autocomplete="off"` at all - it is completely useless these days.

True. I would be open to a Pull Request to remove the autocomplete code from the plugin 👍 

> What a waste of time this whole thing is :)

Haha, agreed.

**Comment by jackocnr** [comment]
Hmm strangely, I've just re-read the [link](https://developer.mozilla.org/en-US/docs/Web/Security/Securing_your_site/Turning_off_form_autocompletion) posted in the opening comment in this issue, and now there is no mention of the random string / "nope" value, it just says setting it to "off" is the standard way and it should work. And again, similar to [last time I tried this](https://github.com/jackocnr/intl-tel-input/issues/668#issuecomment-396070924) back in 2018, setting it to "off" is working for me, in that it prevents autocomplete on Chrome v80.0.3987.149 on MacOS.

So for these reasons, I no longer want to remove it from the plugin - apologies for the change of direction there.

But what we can and should do, is respect any autocomplete attribute that is already present on the input. So in the plugin code, we should only add `autocomplete=off` to the input if it doesn't already have an autocomplete attribute. Similar to the "polite" setting on the `autoPlaceholder` option. And I would welcome a Pull Request that implements this. Thanks.

**Comment by justjuank** [comment]
> Hmm strangely, I've just re-read the [link](https://developer.mozilla.org/en-US/docs/Web/Security/Securing_your_site/Turning_off_form_autocompletion) posted in the opening comment in this issue, and now there is no mention of the random string / "nope" value, it just says setting it to "off" is the standard way and it should work. And again, similar to [last time I tried this](https://github.com/jackocnr/intl-tel-input/issues/668#issuecomment-396070924) back in 2018, setting it to "off" is working for me, in that it prevents autocomplete on Chrome v80.0.3987.149 on MacOS.

What about this part in the link you mentioned?

> If a browser keeps on making suggestions even after setting autocomplete to off, then you have to change the name attribute of the input element.

That is the main issue I think, if you had previous auto suggestions displayed on a input with a specific name, say "phone" (on a website other than yours), and then you use the same input name on a control that you expect not to have auto suggestions displayed, even if you set autocomplete to off it will still display them anyways. That's why my workaround worked, because it dealt with the name attribute and not with the autocomplete attribute since it's kinda useless.

**Comment by jackocnr** [comment]
But we don't touch the name attribute, so that's completely up to the dev, right?

**Comment by justjuank** [comment]
Yeah that's right. Sorry I'm not very familiar with your plugin, but if the browser auto suggestions are being a problem I would at least mention it on the documentation so devs are aware, otherwise don't mind me. I guess the native auto suggestions are a bigger issue for other libraries, they've been a bother to me in the past at least haha.

**Comment by jackocnr** [comment]
Ok we now only set autocomplete=off if it's not already set. Released in v16.0.1.
